solutions/src/main/java/ru/sberbank/school/task06/CollectionUtils.java

```diff
@@ -0,0 +1,185 @@
+     * Проверка на содержение хотя бы одного элемента одной коллекции в другой.
+     *
+     * @param c1 первая коллекция
+     * @param c2 вторая коллекция
+     * @return true, если хотя бы 1 элемент коллекции c2 содержатся в c1
+     */
+    public static <T> boolean containsAny(List<?> c1, List<?> c2) {
+        for (Object element : c2) {
+            if (c1.contains(element)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Создание коллекции, содержащей элементы из входной коллекции в диапазоне от min до max (не
+     * индексы, а значения). Элементы необходимо сравнивать, через интерфейс Comparable Пример:
+     * range(Arrays.asList(8, 1, 3, 5, 6, 4), 3, 6) вернет {3,4,5,6}
+     *
+     * @param list исходная коллекция
+     * @param min  минимальное значение
+     * @param max  максимальнео значение
+     * @return отфильрованный по минимальному и максимальному значению список
+     */
+    public static <T extends Comparable<? super T>> List<T> range(List<? extends T> list, T min, T max) {
+         List<T> newlist = new ArrayList<>();
+         for (T element : list) {
+             if (element.compareTo(min) >= 0 && element.compareTo(max) <= 0) {
+                 if (!newlist.isEmpty()) {
+                     int index = newlist.size();
+                     if (element.compareTo(newlist.get(index-1))<0) {
+                         index = findIndex(newlist, element);
+                     }
+                     newlist.add(index, element);
+                 }  else {
+                     newlist.add(element);
+                 }
+             }
+         }
+         return newlist;
+    }
+
+    private static <T extends Comparable<? super T>> int findIndex(List<T> newlist, T element) {
+        int index = 0;
+        for (T element1 : newlist) {
+            if (element.compareTo(element1) <= 0) {
+                index = newlist.indexOf(element1);
+                break;
+            }
+        }
+        return index;
+    }
+
+
+    /**
+     * Создание коллекции, содержащей элементы из входной коллекции в диапазоне от min до max (не
+     * индексы, а значения). Элементы необходимо сравнивать, через интерфейс Comparator Пример:
+     * range(Arrays.asList(8, 1, 3, 5, 6, 4), 3, 6) вернет {3,4,5,6}
+     *
+     * @param list исходная коллекция
+     * @param min  минимальное значение
+     * @param max  максимальнео значение
+     * @return отфильрованный по минимальному и максимальному значению список
+     */
+    public static <T>List<T> range(List<? extends T> list, T min, T max, Comparator<? super T> comparator) {
+        List<T> newlist = new ArrayList<>();
+        for (T element : list) {
+            if (comparator.compare(element, min) >= 0 && comparator.compare(element, max) <= 0) {
+                if (!newlist.isEmpty()) {
+                    int index = 0;
+                    if (comparator.compare(element, newlist.get(newlist.size()-1))<0) {
```

Comment:
WhitespaceAround: &#39;&lt;&#39; is not followed by whitespace.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a-896477435>*